db/migrate/20191008123620_add_foreign_key_for_messages_and_comments.rb

```diff
@@ -0,0 +1,13 @@
+class AddForeignKeyForMessagesAndComments < ActiveRecord::Migration[5.2]
```

Comment:
Messages may not have an `author_user_id` pointing to the `users` table; it can be `0` to indicate the message is from the system. (See `Message.author_username`.)

Reply:
It is confusing for me, in this case, maybe it needs to add a partial index, but MySQL doesn't support this kind of index. May someone gives advice?

Reply:
Maybe someone can give me a piece of advice on how to resolve this issue ```fail Message author model should have proper foreign key in the database``` in such case?